tsconfig.json

```diff
@@ -3,6 +3,7 @@
   "include": ["./src/**/*.ts"],
   "exclude": ["./src/**/*.cy.ts"],
   "compilerOptions": {
+    "allowImportingTsExtensions": true,
```

Comment:
This does not work, just do not import with `.ts` extension in Typescript files (in JS files it work).

```suggestion
```

Reply:
This works, I missed that styleguidist uses a different config

Reply:
We have 46 places with `.ts` imports in this repo

Reply:
> Importing with `.ts` extension in Typescript is invalid when emitting is enabled, as the output will be invalid.
> So as long as we use a bundler everything works, but e.g. `tsc` would not work, because Typescript does **not** rewrite import extensions, so you either have to import Typescript files with `.js` or without any extension.

We almost never use `tsc` as a compiler (except `notify_push`) and in general, in modern TypeScript usage it's rarely used as a compiler/language and not as a static analyzer (type checker).

Specifically, in this repo, we already had `@babel/typescript-preset` to work with TS as a static analyzer, and only the webpack config was not correct.